lib/utils.js

```diff
@@ -279,6 +279,51 @@ function extend(a, b, thisArg) {
   return a;
 }
 
+function mergeConfig(defaults, instanceConfig) {
+  var config = {};
+  forEach(['url', 'method', 'params', 'data'], function(prop) {
+    config[prop] = instanceConfig[prop];
+  });
+  forEach(['headers', 'auth', 'proxy'], function(prop) {
+    if (!isUndefined(instanceConfig[prop])) {
+      if (!isObject(instanceConfig[prop])) {
+        config[prop] = instanceConfig[prop];
+      } else {
+        config[prop] = merge(
+          defaults[prop],
+          instanceConfig[prop]
+        );
+      }
+    } else if (!isUndefined(defaults[prop])) {
+      config[prop] = JSON.parse(JSON.stringify(defaults[prop]));
+    }
+  });
+  var remainingProperties = [
+    'baseURL',
+    'transformRequest',
+    'transformResponse',
+    'paramsSerializer',
+    'timeout',
+    'withCredentials',
+    'adapter',
+    'responseType',
+    'xsrfCookieName',
+    'xsrfHeaderName',
+    'onUploadProgress',
+    'onDownloadProgress',
+    'maxContentLength',
+    'validateStatus',
+    'maxRedirects',
+    'httpAgent',
+    'httpsAgent',
+    'cancelToken'
+  ];
+  forEach(remainingProperties, function(prop) {
```

Comment:
modify this line to `... function assignProp(prop)` to pass eslint errors in CI tests (or some other function name of your choosing)